driver-core/src/test/unit/com/mongodb/internal/connection/AbstractServerDiscoveryAndMonitoringTest.java

```diff
@@ -105,10 +107,10 @@ protected void applyApplicationError(final BsonDocument applicationError) {
 
         switch (when) {
             case "beforeHandshakeCompletes":
-                server.invalidate(errorGeneration);
+                server.invalidate(BEFORE_HANDSHAKE, exception, errorGeneration, maxWireVersion);
```

Comment:
Now we use the same invalidate method for both cases, so updating this test to match